### ISSUE TITLE
Is the game running faster than it’s supposed to?

### ISSUE THREAD

**Issue by slidelljohn** [VitorVilela7/SA1-Root#5]
Is the game running faster than it’s supposed to?

I seen some comments that you made on the gradius sa-1 mod and it’s seems like you have the sa-1 as the main cpu that calls the snes cpu when needed. Is this correct? If this is true, I’m wondering if it would be better to have the snes cpu as the main cpu that calls the sa-1 so gameplay won’t be faster than what it’s supposed to be like pressing the A button for arcade. I didn’t do a lot of testing but gameplay seems too fast. I could be wrong though. With my sa-1 mod I have the snes cpu as the main cpu that calls the sa-1 for certain functions. 

I have lots of questions but I couldn’t find a way to pm you.

**Comment by VitorVilela7** [comment]
It's not really running faster than supposed. Probably some bosses were however designed to run slight faster to compensate the game slowdown, but in the entire playthough the wing has the same fire rate as it's supposed to be, so it's not really a bug.

Some people managed to finish the game in Arcade mode, so it's not so hard to the impossible level.

The model you proposed is actually the one that I follow on the game.

**Comment by slidelljohn** [comment]
Ok, cool. It was just a few things I was thinking about. All of your work is impressive keep it up. If you ever need any help with anything you know where to find me.

**Comment by VitorVilela7** [comment]
Ah slidelljohn, feel free to contact me at vitor.vilela.mario on the gmail if you have any more questions.

I saw you tried optimizing Gradius III to use FastROM and it was unfortunate on my part for releasing a SA-1 ROM hack of it while you were working on it -- I didn't know any of your work until someone pointed it out after I released the stable patch. Since you have some interest on the SA-1 ROM hacking, I can give you a hand as well if you would like :) but don't give it up please!

**Comment by slidelljohn** [comment]
I was doing the sa-1 before I started the fastrom and since I had the fully documented gradius 3’s asm that I uploaded on romhacking.net I did the fastrom because it was easy to do with those docs. I last worked on my sa-1 in October of last year but I’m going to finish after I do more optimizations. I have a basic setup in my sa-1 and I did a test with it for decompressing function call and it was working but I still have a lot of things to test with it. But yea my journey will continue. I’m not going to stop working on it. The main goal for me with the sa-1 chip is to implement it into mega man x and add the 128kb of vram that my mod chip supports. I was just using gradius as practice since it was a smaller game. I definitely appreciate the offer and I’m sure I’ll be talking to you soon. Thanks Vitor!
